tempesta_fw/connection.h

```diff
@@ -209,9 +210,11 @@ enum {
 typedef struct {
 	TfwCliConn	cli_conn;
 	TlsCtx		tls;
+	TfwHttp2Ctx	*h2;
```

Comment:
The structure name becomes misleading, now it's more than TLS connection, it also stores the HTTP2 context. And a new issue is introduced: HTTP2 context is not allocated for non-TLS HTTP2 connections (which are allowed by standard).
In the same time HTTP2 connections will have two contexts in the same time: HTTP2 and HTTP1. HTTP1 context is stored as `TfwConn->TfwHttpParser`. Probably we should introduce a union in `TfwConn` class to store 'protocol context' whatever http2 or http1. Seems like we never want both. Even is the HTTP1 connection is upgraded to HTTP2 we can easily re-init the field.

Reply:
>HTTP2 context is not allocated for non-TLS HTTP2 connections (which are allowed by standard)

In result of discussion in #309 and in chat - it has been decided not to support non-TLS HTTP/2 in TempestaFW.
>Probably we should introduce a union in TfwConn class to store 'protocol context' whatever http2 or http1.

Yes, agree with that. The similar approach is proposed in https://github.com/tempesta-tech/tempesta/issues/309#issuecomment-467447205 (in `Stream Priority/Dependencies`). These changes is part of future PRs in context of #309 (`"Request HTTP/2 -> HTTP/1.1"` or more sooner `"Disable @seq_queue for HTTP/2"`). During implementation of Framing, Scheduler and Stream States - I temporary left that part as is - for testing/debugging purposes.